### PR TITLE
Adding search by criteria and order by

### PR DESCRIPTION
**Pull request by jdiazgon**
Addressing #14. This pull request implements two features that are needed for our devon4ng client:

- Implements order by: Now it is possible to order your data by any property (asc or desc).
- Implements search by criteria: Now it is possible to retrieve a list of items that match a search.

### CODE REVIEW

File: template/src/todo/todo.controller.ts [hunk 1]
```diff
@@ -32,9 +35,23 @@ export class TodoController {
   @ApiResponse({ status: HttpStatus.OK, type: TodoDTO })
   @ApiResponse({ status: HttpStatus.BAD_REQUEST, type: ApiException })
   @ApiOperation(getOperationId('Todo', 'GetAll'))
-  async getTodos(): Promise<TodoDTO[]> {
+  async getTodos(@Query() params?: any): Promise<TodoDTO[]> {
     try {
-      return plainToClass(TodoDTO, await this._todoService.findAll(), {
+      let filter: FindManyOptions<ObjectLiteral> = {};
+      let orderValue, whereValue: any;
```
Comment (sjimenez77) [inline template/src/todo/todo.controller.ts]: We should remove `any` types as much as we can. I prefer to use union types as `string | number` in order to consider alternatives that we already know.

**Comment by sjimenez77** [comment]
Hi @jdiazgon,

Although this is a different topic, could you please remove in this PR the asciidoc files from root folder as they are already in the `.github` folder? I mean the contributing guide, templates, etc. 

Thanks!

**Comment by dario-rodriguez** [comment]
Hello @jdiazgon,
IMHO the approach using the query instead the body is the correct. The problem is that your solution only works for typeorm. I think that our solution should work with other ORM (e.g. mongoose).
Anyway, if you need to continue with your work before we discussed it, you can use this approach or the approach that I mentioned in the issue, as you want.

Kind regards

**Comment by dario-rodriguez** [comment]
And I miss something important: the todo-app is an old example that will be updated (probably removed) soon.

**Comment by dario-rodriguez** [comment]
Corrected in devon4node 1.0.0. This PR is not needed

**Pull request closed without merging**